### ISSUE TITLE
Simple table with no primary key raises error about missing name not key

### ISSUE THREAD

**Issue by davidism** [mitsuhiko/flask-sqlalchemy#552]
Due to the way `__table_cls__` is deciding to create a table, we get the wrong error message when a tablename is generated for a non-inheriting model with no primary key.

```python
class User(db.Model):
    pass
```

Error is:

    sqlalchemy.exc.InvalidRequestError: Class <class '__main__.User'> does not have a __table__ or __tablename__ specified and does not inherit from an existing table-mapped class.

Error should be:

    sqlalchemy.exc.ArgumentError: Mapper Mapper|User|user could not assemble any primary key columns for mapped table 'user'

**Comment by davidism** [comment]
Possible solution: add a check after primary key loop to see if a base class is a mapped model and if not, create the table anyway, instead of unconditionally deleting the name.

**Comment by JacobSMoller** [comment]
Running 2.3.0 i get the `does not have a __table__ or __tablename__` even though my class looks like:

```class Account(db.Model):
         __tablename__ = 'account'
```

**Comment by davidism** [comment]
The only way that could have worked is if the table `account` was already defined, such as from reflection. See #550, fixed by #551.

You have to have one of the following: an existing table with that name, a primary key, or an inherited model.

**Comment by JacobSMoller** [comment]
Its reflected, will look at it after #551 is released. Thanks a ton for the quick answer and your work on this project.
